GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/DehydrateTests.cs

```diff
@@ -395,19 +395,32 @@ public void FolderDehydrateCannotDehydrateDotGitFolder()
             this.DehydrateShouldSucceed(new[] { $"Cannot dehydrate folder '{TestConstants.DotGit.Info.Root}': invalid folder path." }, confirm: true, noStatus: false, foldersToDehydrate: TestConstants.DotGit.Info.Root);
         }
 
-        [TestCase]
-        public void FolderDehydratePreviouslyDeletedFolder()
+        [TestCase]
+        public void FolderDehydratePreviouslyDeletedFolders()
         {
             string folderToDehydrate = "TrailingSlashTests";
             TestPath folderToDelete = new TestPath(this.Enlistment, folderToDehydrate);
+            string secondFolderToDehydrate = "FilenameEncoding";
+            TestPath secondFolderToDelete = new TestPath(this.Enlistment, secondFolderToDehydrate);
+
             this.fileSystem.DeleteDirectory(folderToDelete.VirtualPath);
-            GitProcess.Invoke(this.Enlistment.RepoRoot, "commit -a -m \"Delete a directory\"");
+            this.fileSystem.DeleteDirectory(secondFolderToDelete.VirtualPath);
+            GitProcess.Invoke(this.Enlistment.RepoRoot, "commit -a -m \"Delete directories\"");
 
             GitProcess.Invoke(this.Enlistment.RepoRoot, "checkout -f HEAD~1");

```

Comment:
I wonder if we should check for the folder existence before and after the `checkout`?